crates/router/src/connector/expresscheckout/transformers.rs

```diff
@@ -0,0 +1,363 @@
+    amount: String,
+    #[serde(rename = "order.return_url")]
+    return_url: String,
+    #[serde(rename = "order.currency")]
+    currency: String,
+    #[serde(rename = "order.gateway_id")]
+    gateway_id: u8,
+    #[serde(flatten)]
+    gateway_reference_metadata: HashMap<String, String>,
+    merchant_id: String,
+    payment_method_type: PaymentMethodType,
+    card_number: String,
+    card_exp_month: String,
+    card_exp_year: String,
+    name_on_card: String,
+    card_security_code: String,
+    format: String,
+    save_to_locker: bool,
+}
+
+#[derive(Serialize, Debug, Default, Eq, PartialEq)]
+enum PaymentMethodType {
+    #[default]
+    Card,
+}
+
+// #[derive(Default, Debug, Eq, PartialEq, Serialize)]
+// pub struct OrderDetails {
+//     order_id: String,
+//     amount: String,
+//     return_url: String,
+//     currency: String,
+//     gateway_id: u8
+// }
+
+#[derive(Deserialize, Debug)]
+pub struct GatewayMetadata {
+    gateway_id: u8,
+    gateway_reference_key: String,
+    gateway_reference_value: String,
+}
+
+impl TryFrom<&types::PaymentsAuthorizeRouterData> for ExpresscheckoutPaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
+        let metadata = item.connector_meta_data.clone().ok_or(
+            errors::ConnectorError::MissingRequiredField {
+                field_name: ("metadata"),
+            },
+        )?;
+        let gateway_metadata: GatewayMetadata = serde_json::from_value(metadata)
+            .map_err(|_| errors::ConnectorError::NoConnectorMetaData)?;
+        let mut gateway_reference_key_prefix = String::from("order.");
+        gateway_reference_key_prefix.push_str(&gateway_metadata.gateway_reference_key);
+        let gateway_reference_metadata = HashMap::from([(
+            gateway_reference_key_prefix,
+            gateway_metadata.gateway_reference_value,
+        )]);
+        match item.request.payment_method_data {
+            api::PaymentMethod::Card(ref ccard) => {
+                let return_url: String = item
+                    .return_url
+                    .clone()
+                    .ok_or_else(utils::missing_field_err("return_url"))?;
+                Ok(Self {
+                    order_id: item.payment_id.clone(),
+                    amount: item.request.amount.to_string(),
+                    return_url,
+                    currency: item.request.currency.to_string().to_uppercase(),
+                    gateway_id: gateway_metadata.gateway_id,
+                    gateway_reference_metadata,
+                    merchant_id: item.merchant_id.clone(),
+                    payment_method_type: PaymentMethodType::Card,
+                    card_number: ccard.card_number.peek().clone(),
+                    card_exp_month: ccard.card_exp_month.peek().clone(),
+                    card_exp_year: ccard.card_exp_year.peek().clone(),
```

Comment:
We accept expiry year in two formats 25 or 2025. Please make sure connector also supports the same or else we can handle both the cases here and send the field in the proper format.